### ISSUE TITLE
dry-validation does not allow `none?`

### ISSUE THREAD

**Issue by theozaurus** [apotonick/reform#381]
I may be missing something obvious here, but I cannot understand why this dry-validation (0.9.2) schema does not give the same results as a reform (2.2.1) form built in the same way:

```ruby
schema = Dry::Validation.Form do
  required(:name).maybe(:filled?)
end

expect(schema.call(name: nil)).to be_a_success
expect(schema.call(name: 'works')).to be_a_success
expect(schema.call(name: '')).to be_a_success
expect(schema.call(name: '').output).to eq(name: nil) # Form schema converts '' to nil

form = Class.new(Reform::Form) do
  feature Reform::Form::Dry

  property(:name)

  validation do
    required(:name).maybe(:filled?)
  end
end

expect(form.new(OpenStruct.new).validate(name: nil)).to be(true) # Fails
expect(form.new(OpenStruct.new).validate(name: 'works')).to be(true) # Correct
expect(form.new(OpenStruct.new).validate(name: '')).to be(true) # Correct
```

Is this a bug, or have I miss-understood how things work?

Thanks,

Theo

_(updated as I realised Reform uses the [Form](http://dry-rb.org/gems/dry-validation/forms/) schema not the Schema schema from dry-rb)_

**Comment by theozaurus** [comment]
Digging in it appears that the `OpenStruct.new(name: nil)` when run through Disposable it cuts out the `name` attribute altogether, rather than keeping it nil.

**Comment by theozaurus** [comment]
Works with experiment on disposable gem as discussed on Gitter:

```
# Gemfile
gem 'disposable', require: false, github: 'apotonick/disposable', branch: 'render_nil'
```

**Comment by apotonick** [comment]
I've released disposable-0.4.0. :wink:

**Comment by theozaurus** [comment]
Thanks!
